### ISSUE TITLE
comat模块问题

### ISSUE THREAD

**Issue by 66D9** [zjukg/NATIVE#11]
作者您好，我想复现您消融实验去掉comat模块的部分，我修改了train_one_step方法：

![Image](https://github.com/user-attachments/assets/8479fd31-4b6b-470e-8f99-3c7d928f230c)
但是我发现实验结果几乎没有变化，下面是在mkgw数据集上训练500轮的测试结果，第一张图是去掉comat的结果：

<img width="619" height="112" alt="Image" src="https://github.com/user-attachments/assets/3a556b1e-b0d5-41c2-b346-10ea02c394bd" />

<img width="640" height="91" alt="Image" src="https://github.com/user-attachments/assets/99d1ab6f-90e4-4996-a2b0-21c0122273f4" />
另外，我在mkgy数据集上也测试了1000轮的结果效果也一样，请问是我去掉comat模块时修改代码的问题吗

**Comment by Zhang-Each** [comment]
这个模块对不同的数据集的增益效果有差别，它需要在一部分有效的或者说高质量的模态信息引导下，通过GAN来对低质量的多模态信息做增强，而MKG-W和MKG-Y本身的多模态信息质量较低，用GAN做增强的效果有限。另外就是实验环境、参数配置和代码可能有所差别。

**Comment by zhangyushinibabadebaba** [comment]
你好，你现在复现成功了吗，我做消融实验在DB15K上面，我发现实验结果也没有变化，代码和你差不多。

**Comment by 66D9** [comment]
> 你好，你现在复现成功了吗，我做消融实验在DB15K上面，我发现实验结果也没有变化，代码和你差不多。

没有😁

**Comment by Zhang-Each** [comment]
> > 你好，你现在复现成功了吗，我做消融实验在DB15K上面，我发现实验结果也没有变化，代码和你差不多。
> 
> 没有😁

麻烦你把完整的代码贴一下，可以排查下哪里有问题。我重复了一下DB15K上的实验发现结果和论文表里写的一致。

**Comment by Zhang-Each** [comment]
你好，我在自己的实验环境下测得MKG-W和MKG-Y上的消融实验结果分别为

(MKG-W) averaged(filter):	 0.331492 	 769.339233 	 0.464202 	 0.362424 	 0.259827 

(MKG-Y) averaged(filter):	 0.297666 	 1332.739624 	 0.413444 	 0.329140 	 0.232257 

实验中dim=200，neg_num=64，epoch=1000，我们论文里MKG-W和MKG-Y的实验结果也是用相同的参数训练得到的。

**Comment by 66D9** [comment]
我用的4090，我只修改了图中的部分，还有训练过程中生成器的损失删除掉了，其它代码都没有修改。        
for epoch in training_range:
            res = 0.0
            res_g = 0.0
            for data in self.data_loader:
                loss = self.train_one_step(data)
                res += loss
还有我用的你脚本中设置的neg_num=128，dim=250和你用的不太一样。但是我尝试过neg_num=64，效果也是一样的

**Comment by Zhang-Each** [comment]
> 我用的4090，我只修改了图中的部分，还有训练过程中生成器的损失删除掉了，其它代码都没有修改。 for epoch in training_range: res = 0.0 res_g = 0.0 for data in self.data_loader: loss = self.train_one_step(data) res += loss 还有我用的你脚本中设置的neg_num=128，dim=250和你用的不太一样。但是我尝试过neg_num=64，效果也是一样的

对MKG的两个数据集我们的实验都是在dim=200下完成的，你提到的情况我之前也从来没遇到过，我在MKG-W和MKG-Y上重复了多次实验，去掉CoMAT模块性能都出现明显下降，实验结果和我上面展示的结果一致。根据已有信息，我目前也找不出导致你这一情况的原因。

**Comment by 66D9** [comment]
我使用的代码时这个项目上你开源的代码，要不你试试使用这个项目上你的默认配置和代码？

**Comment by Zhang-Each** [comment]
我实验用的代码和仓库提供的代码是一致的，仓库里提供的shell脚本并不是论文中的实验所使用的参数。消融实验的结果也如上。

**Comment by Zhang-Each** [comment]
在dim=250，neg_num=128的情况下得到的结果：

MKG-W averaged(filter):	 0.329636 	 781.437256 	 0.463383 	 0.361488 	 0.256902
MKG-Y  averaged(filter):	 0.294315 	 1354.375000 	 0.414758 	 0.323695 	 0.226624
